File: JavaScript Project/SimpleAndCompoundInterestCalculator.js
function calculateSimpleInterest(principal, rate, time) {
    // Simple Interest formula: SI = P * R * T / 100
    const simpleInterest = (principal * rate * time) / 100;
    return simpleInterest;
}

function calculateCompoundInterest(principal, rate, time) {
    // Compound Interest formula: CI = P * (1 + (R / 100))^T - P
    const compoundInterest = principal * (Math.pow((1 + rate / 100), time)) - principal;
    return compoundInterest;
}

function main() {
    try {
        const principal = parseFloat(prompt("Enter the principal amount: "));
        const rate = parseFloat(prompt("Enter the annual interest rate: "));
        const time = parseFloat(prompt("Enter the time period in years: "));

        const simpleInterest = calculateSimpleInterest(principal, rate, time);
        const compoundInterest = calculateCompoundInterest(principal, rate, time);

        console.log(`\nSimple Interest: ${simpleInterest.toFixed(2)}`);
        console.log(`Compound Interest: ${compoundInterest.toFixed(2)}`);
    } catch (error) {
        console.error("Invalid input. Please enter numeric values for principal, rate, and time.");
    }
}

main();

File: Python Project/SimpleAndCompoundInterestCalculator.py
def calculate_simple_interest(principal, rate, time):
    # Simple Interest formula: SI = P * R * T / 100
    simple_interest = (principal * rate * time) / 100
    return simple_interest

def calculate_compound_interest(principal, rate, time):
    # Compound Interest formula: CI = P * (1 + (R / 100))^T - P
    compound_interest = principal * (1 + rate / 100)**time - principal
    return compound_interest

def main():
    try:
        principal = float(input("Enter the principal amount: "))
        rate = float(input("Enter the annual interest rate: "))
        time = float(input("Enter the time period in years: "))

        simple_interest = calculate_simple_interest(principal, rate, time)
        compound_interest = calculate_compound_interest(principal, rate, time)

        print(f"\nSimple Interest: {simple_interest:.2f}")
        print(f"Compound Interest: {compound_interest:.2f}")
    except ValueError:
        print("Invalid input. Please enter numeric values for principal, rate, and time.")

#calling the main function directly
main()
